### PR TITLE
[1/5] Add Travis Continuous Integration

### LINKED ISSUE #5: Setup CI
CI should:
- Check all unit tests pass
- Check code is properly formatted
- Check example compiles
- Other stuff?

### PR DESCRIPTION
**Pull request by nlordell**
Closes #5

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  fast_finish: true
+
+cache: cargo
+before_cache:
+  - rm -rf "$TRAVIS_HOME/.cargo/registry/src"
+
+before_install:
+  - rustup component add clippy rustfmt
+install:
+  - (cd contracts; npm install)
```
Comment (fleupold) [inline .travis.yml:18]: nit: node modules could be cached as well.

File: src/transaction.rs [hunk 2]
```diff
@@ -1,6 +1,9 @@
 //! Implementation for setting up, signing, estimating gas and executing
 //! transactions on the Ethereum network.
 
+#![allow(clippy::large_enum_variant)]
```
Comment (fleupold) [inline src/transaction.rs:4]: Is it possible to allow this only for the enum/type that large/complex?

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2F3o7WTJMjG44pEebcJy%2Fgiphy.gif&token=continuous_integration" data-gitmeme-token="continuous_integration"></a>

File: src/transaction.rs (hunk 2 above)
Comment (nlordell) [reply src/transaction.rs:4]: We address this in a later commit, I just wanted this to get through the CI for now.

**Pull request merged**